tests/collective_ops/test_sendrecv.py

```diff
@@ -104,3 +104,23 @@ def test_sendrecv_scalar_jit():
 
     assert jnp.array_equal(res, jnp.ones_like(arr) * other)
     assert jnp.array_equal(_arr, arr)
+
+
+@pytest.mark.skipif(size < 2 or rank > 1, reason="Runs only on rank 0 and 1")
+def test_sendrecv_grad():
+    from mpi4jax import sendrecv
+
+    arr = jnp.ones((3, 2)) * (rank + 1)
+    _arr = arr.copy()
+
+    other = 1 - rank
+
+    def f(x):
+        x, token = sendrecv(x, x, source=other, dest=other)
```

Comment:
Maybe have a test where `x` is also manipulated *before* calling `sendrecv`, so the transmitted gradient isn't just 1.